_drafts/release-workflows.md

```diff
@@ -0,0 +1,388 @@
+| **merge** some commits from `next` branch to `master` | Get the last release for commits on branch `master` (including the one that come from `next`), analyze commits, increase the version; If the commit associated with the last release found is also present on `next`, make the version available on the `latest` channel, otherwise do a regular release. |
+
+##### Future branches configuration
+
+```json
+"release": {
+  "branches": [
+    "master", {
+      "branch": "next",
+      "type": "major"
+    }
+  ]
+}
+```
+
+The `type` is required for future branches and indicate that its reserved for a certain type of release (`minor` or `major`).
+That defines how much ahead the future branch must be versus the pervious branch in the list. If configured with `major`, the `next` branch must always one `major` release ahead of `master`.
+
+Example 1:
+- if the last release on `latest` is `2.1.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is also `2.1.0`
+- then any releases are allowed on `latest`
+- And only `major` releases are allowed on `next`
+
+Example 2:
+- if the last release on `latest` is `2.1.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is `2.5.0`
+- then `patch` releases are allowed on `latest`, but `minor` and `major` are forbidden
+- And any releases are allowed on `next`
+
+Example 3:
+- if the last release on `latest` is `2.1.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is `3.0.0`
+- then `patch` and `minor` releases are allowed on `latest`, but `major` are forbidden
+- And any releases are allowed on `next`
+
+Example 4:
+- if the last release on `latest` is `3.0.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is `2.5.0`
+- And any releases are allowed on `latest`
+- No releases are allowed on `next` (it's considered stalled) until `master` is merged into `next`. Once it's done the situation becomes the one described in Example 1
+
+That enforces consistence across versions: **if a `x.y.z` version exists on a given channel, all superior versions on that channel must includes all the commits of `x.y.z`**
+
+If not specified the `type` is `major`.
+
+With specific channel names:
+
+```json
+"release": {
+  "branches": [
+    "master", 
+    {
+      "branch": "next",
+      "channel": "experimental",
+      "type": "major"
+    }
+  ]
+}
+```
+
+#### LTS branches
+
+Workflow to release legacy versions.
+
+Ideal for modules maintaining legacy versions and doing bug fixes and features only releases.
+
+Branch: `1.x.x`, `2.x.x`, `master` (any number of lts branches are supported)
```

Comment:
yeah I know, but I think I’d start with more restrictive branch name conventions and don’t support `1.x.x` or `^1.0.0` as branch names, just to keep things simple

Reply:
That's actually more simple to support all, because we can do `semver.validRange` and we would not have to deal with custom regex to guarantee that only the form `1.x` is used.

Reply:
I know, I don’t mean simple to implement, but simple to understand and document

Reply:
Well, in the doc I agree we should mention only the `1.x` form. Getting in the details I went here is not necessary for the doc.

Reply:
another reason to only support the `1.x` format is that we don’t have to worry about overlapping ranges, like with `1.x` and `1.1.x`, for example

Reply:
are there any of these formats that would conflict with tags created for the versions that get released?